app/services/amr/n3rgy_readings_download_and_upsert.rb

```diff
@@ -57,10 +52,11 @@ def create_import_log(start_date, end_date)
 
     def current_date_range_of_readings
       if existing_n3rgy_readings.any?
-        first = existing_n3rgy_readings.minimum(:reading_date)
-        last = existing_n3rgy_readings.maximum(:reading_date)
-        (n3rgy_first_reading_of_day(DateTime.parse(first + 'T00:30'))..
-          n3rgy_last_reading_of_day(DateTime.parse(last + 'T00:00')))
+        first = DateTime.parse(existing_n3rgy_readings.minimum(:reading_date))
```

Comment:
Assume it's ok that you've lost the `+ 'T00:30'` etc from the params to DateTimeParse here?